src/views/smart_table.rs

```diff
@@ -0,0 +1,219 @@
+                            self.limiters[i] = prev_limiter + CELL_MIN_SIZE_PX;
+                        }
+                    }
+
+                    // Set new Sizes
+
+                    if i == 0 {
+                        self.sizes[i] = self.limiters[i];
+                    } else {
+                        self.sizes[i] = self.limiters[i] - prev_limiter;
+                    }
+
+                    if i == self.limiters.len() - 1 {
+                        self.sizes[i + 1] = w - self.limiters[i]
+                    } else {
+                        self.sizes[i + 1] = next_limiter - self.limiters[i];
+                    }
+                }
+            }
+
+            WindowEvent::MouseUp(b) => {
+                if *b == MouseButton::Left {
+                    cx.emit(SmartTableEvent::StopDrag)
+                }
+            }
+            _ => {}
+        });
+    }
+}
+
+pub struct SmartTableRow {}
+
+impl SmartTableRow {
+    pub fn new<'a, L>(
+        cx: &'a mut Context,
+        data: &'a Vec<&'a str>,
+        sizes: L,
+        header: bool,
+    ) -> Handle<'a, Self>
+    where
+        L: Lens<Target = Vec<f32>>,
+    {
+        Self {}
+            .build(cx, |cx| {
+                for (i, d) in data.iter().enumerate() {
+                    if i != 0 {
+                        let element =
+                            Element::new(cx).class("smart-table-divisor").class("vertical");
+                        if header {
+                            element.class("accent");
+                            ResizeHandle::new(cx, i - 1, true)
+                                .toggle_class("active", SmartTable::dragging.map(|d| d.is_some()));
+                        }
+                    }
+
+                    HStack::new(cx, |cx| {
+                        Label::new(cx, *d).hoverable(false);
+                    })
+                    .hoverable(false)
+                    .class("smart-table-row-data-container")
+                    .width(sizes.clone().map(move |v| {
+                        if v[i] == 0.0 {
+                            Stretch(1.0)
+                        } else {
+                            Pixels(v[i])
+                        }
+                    }));
+                }
+            })
+            .toggle_class("header", header)
+            .layout_type(LayoutType::Row)
+    }
+}
+
+impl View for SmartTableRow {
+    fn element(&self) -> Option<&'static str> {
+        Some("smart-table-row")
+    }
+}
+
+pub struct ResizeHandle;
+
+impl ResizeHandle {
+    pub fn new(cx: &mut Context, i: usize, vertical: bool) -> Handle<Self> {
+        Self {}
+            .build(cx, |cx| {
+                Element::new(cx)
+                    .position_type(PositionType::SelfDirected)
+                    .class("resize-handle")
+                    .cursor(CursorIcon::EwResize)
+                    .on_press_down(move |cx| cx.emit(SmartTableEvent::StartDrag(i)));
```

Comment:
This could be moved to the `ResizeHandle` itself rather than having an inner element.

Reply:
True, but that way we can't control how width we'd like the handle to be. The handle itself is wider than the divisor so the whole header section would have an offset. This way the layout system doesn't notice there is a handle at all and we can make it more accessible.

Is that exact behavior the supposed one? Lemme know.

Reply:
Not sure I understand, if you make the handle itself self-directed would it not have the same effect?